src/java.base/share/classes/java/text/ChoiceFormat.java

```diff
@@ -176,6 +176,8 @@ public class ChoiceFormat extends NumberFormat {
      * @param newPattern See the class description.
      * @throws    NullPointerException if {@code newPattern}
      *            is {@code null}
+     * @throws    IllegalArgumentException if {@code newPattern}
+     *            is invalid
```

Comment:
This would probably more informative if it was changed to `IllegalArgumentException if {@code newPattern} violates the ChoiceFormat pattern syntax`.

However, as of now I kept it this way because there is no defined syntax ATM (although I have a PR for that here -> https://github.com/openjdk/jdk/pull/15392) and the other Format classes all also follow the "... is invalid" wording.